### ISSUE TITLE
Stylelint 16.8.2 failing build and exiting with code 2 when there are no errors

### ISSUE THREAD

**Issue by brooklyndippo** [stylelint/stylelint#7984]
### What minimal example or steps are needed to reproduce the bug?

Bumped stylelint from 15.10.2 to stylelint 16.8.2. When I run `stylelint` on the CLI or in build, I have the exact same results: 0 errors and 278 warnings (a lot I know, but they shouldn't prevent it from passing). The branch on 15.10.2 is passing the build and the other one is failing with exit code 2 even though the `Errors` output is completely empty.

On my branch with the updated version of stylelint I am getting `The build command failed with status 2.` and the final container shows `{... ExitCode: 2, Errors: "", ....}` so it has the same empty string as the other one for errors, but in this case it is showing returning exit code 2.

Looking through Github, I found similar issues from several years ago:

#3077 

### What minimal configuration is needed to reproduce the bug?

```json
"stylelint": {
    "extends": [
      "@shopify/stylelint-plugin/prettier",
      "@shopify/stylelint-polaris"
    ],
    "rules": {
      "scss/at-use-no-unnamespaced": null,
      "at-rule-empty-line-before": null,
      "comment-no-empty": null,
      "scss/comment-no-empty": true,
      "scss/at-import-partial-extension": null,
      "scss/load-partial-extension": "never"
    }
  },
  ```

### How did you run Stylelint?

`yarn stylelint './**/*.{css,scss}' --allow-empty-input  --ignore-pattern node_modules`

### Which Stylelint-related dependencies are you using?

```json
"dependencies": {
    ...
    "@shopify/stylelint-plugin": "^14.0.0",
    "@shopify/stylelint-polaris": "^16.0.5",
    "stylelint": "^16.8.2"
 },
 "resolutions": {
    ... 
    "stylelint-scss": "6.5.1"
 }, 
```

### What did you expect to happen?

I expected the build to pass since there were no stylelint errors. 

### What actually happened?

The build failed with exit code 2.

**Comment by ybiquitous** [comment]
@brooklyndippo Thanks for the report.

The following change since 16.0.0 may be related, but I'm not sure.
https://stylelint.io/migration-guide/to-16#changed-cli-exit-code-for-flag-errors

Can you provide a reproducible repository so that we can investigate this issue?

---

FYI, here's the related code in the CLI module:
https://github.com/stylelint/stylelint/blob/462cdab4c08231b310c4d428cafc6eff01051e2b/lib/cli.mjs#L522-L533

**Comment by Mouvedia** [comment]
There are 2 other cases on `meow` side:

- [unknown flags](https://github.com/sindresorhus/meow/blob/87d1bc394d9a1d73b76a81d4874db4f1190259ef/source/validate.js#L75)
- [required flags](https://github.com/sindresorhus/meow/blob/87d1bc394d9a1d73b76a81d4874db4f1190259ef/source/validate.js#L118)

We don't have required flags though.

**Comment by brooklyndippo** [comment]
@ybiquitous thanks for looking into this! I'm not sure exactly what to put in a repo since its not giving me any errors to look into. I was working on it more today trying to figure out why it is returning exit code 2 with no errors and I discovered that if I run the same stylelint command with npx instead of yarn then I don't get exit code 2. 
If I run the following yarn command I get the following response:
` yarn stylelint './**/*.{css,scss}' --allow-empty-input  --ignore-pattern node_modules --ignore-pattern vendor/bundle --ignore-pattern app/assets`

```md
⚠ 278 problems (0 errors, 278 warnings)
error Command failed with exit code 2.
```

But if I run the same command with npx instead I get the expected behavior where no exit code is returned:
`npx stylelint './**/*.{css,scss}' --allow-empty-input  --ignore-pattern node_modules --ignore-pattern vendor/bundle --ignore-pattern app/assets`

```md
⚠ 278 problems (0 errors, 278 warnings)
```

Does that help to debug this issue a bit further? We are on yarn 1.22.19 and npx 10.5.0.

**Comment by Mouvedia** [comment]
Can you try `yarn --verbose …`?

**Comment by Mouvedia** [comment]
@brooklyndippo I am wondering if we are being affected by yarnpkg/yarn#5005?

@ybiquitous is the `${EOL}` placement of a bit weird?
cf https://github.com/stylelint/stylelint/commit/571fde1b0be51f00eead68a6bf66757564591afc#diff-347ff93ed2b00c93c817863e32fbac5b4fac71d7339a48378980e682777689f4R512

**Comment by Mouvedia** [comment]
@brooklyndippo I am wondering if we are being affected by yarnpkg/yarn#5005?

**Comment by ybiquitous** [comment]
Umm, I'm still not sure. I want you to provide a minimal reproduction for investigation. I tried it, but sadly I cannot reproduce the issue. 😓 

`test.css`:

```css
a {}
```

`package.json`:

```json
{
  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e",
  "dependencies": {
    "stylelint": "16.9.0"
  },
  "version": "0.0.0"
}
```

`stylelint.config.mjs`:

```js
export default {
	rules: {
		'block-no-empty': [true, {severity: 'warning'}],
	},
};
```

Run:

```sh-session
$ yarn run stylelint '*.css'
yarn run v1.22.22
warning package.json: No license field
$ /Users/masafumi.koba/tmp/foo/node_modules/.bin/stylelint '*.css'

test.css
  1:3  ⚠  Unexpected empty block  block-no-empty

⚠ 1 problem (0 errors, 1 warning)

✨  Done in 0.50s.

$ echo $?
0
```

**Comment by jeddy3** [comment]
Closing as stale. Feel free to reopen with a reproducible example, though.
